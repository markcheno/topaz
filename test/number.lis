#    0: adjs		1
#    5: pushreal	3.1415
#   10: setglobal	0
#   15: pushint		0
#   20: pushstring	'Default'
#   33: pushint		1
#   38: call		println
#   43: pop
#   44: pushint		0
#   49: pushstring	'pi (f)     = %f
'
#   71: getglobal	0
#   76: pushint		2
#   81: call		printf
#   86: pop
#   87: pushint		0
#   92: pushint		0
#   97: call		println
#  102: pop
#  103: pushint		0
#  108: pushstring	'right justified'
#  129: pushint		1
#  134: call		println
#  139: pop
#  140: pushint		0
#  145: pushstring	'pi (7.0f)  = %7.0f
'
#  170: getglobal	0
#  175: pushint		2
#  180: call		printf
#  185: pop
#  186: pushint		0
#  191: pushstring	'pi (7.1f)  = %7.1f
'
#  216: getglobal	0
#  221: pushint		2
#  226: call		printf
#  231: pop
#  232: pushint		0
#  237: pushstring	'pi (7.2f)  = %7.2f
'
#  262: getglobal	0
#  267: pushint		2
#  272: call		printf
#  277: pop
#  278: pushint		0
#  283: pushstring	'pi (7.3f)  = %7.3f
'
#  308: getglobal	0
#  313: pushint		2
#  318: call		printf
#  323: pop
#  324: pushint		0
#  329: pushstring	'pi (7.4f)  = %7.4f
'
#  354: getglobal	0
#  359: pushint		2
#  364: call		printf
#  369: pop
#  370: pushint		0
#  375: pushint		0
#  380: call		println
#  385: pop
#  386: pushint		0
#  391: pushstring	'right justified with negative'
#  426: pushint		1
#  431: call		println
#  436: pop
#  437: pushint		0
#  442: pushstring	'-pi (7.0f)  = %7.0f
'
#  468: getglobal	0
#  473: neg
#  474: pushint		2
#  479: call		printf
#  484: pop
#  485: pushint		0
#  490: pushstring	'-pi (7.1f)  = %7.1f
'
#  516: getglobal	0
#  521: neg
#  522: pushint		2
#  527: call		printf
#  532: pop
#  533: pushint		0
#  538: pushstring	'-pi (7.2f)  = %7.2f
'
#  564: getglobal	0
#  569: neg
#  570: pushint		2
#  575: call		printf
#  580: pop
#  581: pushint		0
#  586: pushstring	'-pi (7.3f)  = %7.3f
'
#  612: getglobal	0
#  617: neg
#  618: pushint		2
#  623: call		printf
#  628: pop
#  629: pushint		0
#  634: pushstring	'-pi (7.4f)  = %7.4f
'
#  660: getglobal	0
#  665: neg
#  666: pushint		2
#  671: call		printf
#  676: pop
#  677: pushint		0
#  682: pushint		0
#  687: call		println
#  692: pop
#  693: pushint		0
#  698: pushstring	'right justified with plus'
#  729: pushint		1
#  734: call		println
#  739: pop
#  740: pushint		0
#  745: pushstring	'pi (+7.0f) = %+7.0f
'
#  771: getglobal	0
#  776: pushint		2
#  781: call		printf
#  786: pop
#  787: pushint		0
#  792: pushstring	'pi (+7.1f) = %+7.1f
'
#  818: getglobal	0
#  823: pushint		2
#  828: call		printf
#  833: pop
#  834: pushint		0
#  839: pushstring	'pi (+7.2f) = %+7.2f
'
#  865: getglobal	0
#  870: pushint		2
#  875: call		printf
#  880: pop
#  881: pushint		0
#  886: pushstring	'pi (+7.3f) = %+7.3f
'
#  912: getglobal	0
#  917: pushint		2
#  922: call		printf
#  927: pop
#  928: pushint		0
#  933: pushstring	'pi (+7.4f) = %+7.4f
'
#  959: getglobal	0
#  964: pushint		2
#  969: call		printf
#  974: pop
#  975: pushint		0
#  980: pushint		0
#  985: call		println
#  990: pop
#  991: pushint		0
#  996: pushstring	'left justified'
# 1016: pushint		1
# 1021: call		println
# 1026: pop
# 1027: pushint		0
# 1032: pushstring	'pi (-7.0f)  = %-7.0f
'
# 1059: getglobal	0
# 1064: pushint		2
# 1069: call		printf
# 1074: pop
# 1075: pushint		0
# 1080: pushstring	'pi (-7.1f)  = %-7.1f
'
# 1107: getglobal	0
# 1112: pushint		2
# 1117: call		printf
# 1122: pop
# 1123: pushint		0
# 1128: pushstring	'pi (-7.2f)  = %-7.2f
'
# 1155: getglobal	0
# 1160: pushint		2
# 1165: call		printf
# 1170: pop
# 1171: pushint		0
# 1176: pushstring	'pi (-7.3f)  = %-7.3f
'
# 1203: getglobal	0
# 1208: pushint		2
# 1213: call		printf
# 1218: pop
# 1219: pushint		0
# 1224: pushstring	'pi (-7.4f)  = %-7.4f
'
# 1251: getglobal	0
# 1256: pushint		2
# 1261: call		printf
# 1266: pop
# 1267: pushint		0
# 1272: pushint		0
# 1277: call		println
# 1282: pop
# 1283: pushint		0
# 1288: pushstring	'left justified with negative'
# 1322: pushint		1
# 1327: call		println
# 1332: pop
# 1333: pushint		0
# 1338: pushstring	'-pi (-7.0f) = %-7.0f
'
# 1365: getglobal	0
# 1370: neg
# 1371: pushint		2
# 1376: call		printf
# 1381: pop
# 1382: pushint		0
# 1387: pushstring	'-pi (-7.1f) = %-7.1f
'
# 1414: getglobal	0
# 1419: neg
# 1420: pushint		2
# 1425: call		printf
# 1430: pop
# 1431: pushint		0
# 1436: pushstring	'-pi (-7.2f) = %-7.2f
'
# 1463: getglobal	0
# 1468: neg
# 1469: pushint		2
# 1474: call		printf
# 1479: pop
# 1480: pushint		0
# 1485: pushstring	'-pi (-7.3f) = %-7.3f
'
# 1512: getglobal	0
# 1517: neg
# 1518: pushint		2
# 1523: call		printf
# 1528: pop
# 1529: pushint		0
# 1534: pushstring	'-pi (-7.4f) = %-7.4f
'
# 1561: getglobal	0
# 1566: neg
# 1567: pushint		2
# 1572: call		printf
# 1577: pop
# 1578: pushint		0
# 1583: pushint		0
# 1588: call		println
# 1593: pop
# 1594: pushint		0
# 1599: pushstring	'left justified with plus'
# 1629: pushint		1
# 1634: call		println
# 1639: pop
# 1640: pushint		0
# 1645: pushstring	'pi (-+7.0f) = %-+7.0f
'
# 1673: getglobal	0
# 1678: pushint		2
# 1683: call		printf
# 1688: pop
# 1689: pushint		0
# 1694: pushstring	'pi (-+7.1f) = %-+7.1f
'
# 1722: getglobal	0
# 1727: pushint		2
# 1732: call		printf
# 1737: pop
# 1738: pushint		0
# 1743: pushstring	'pi (-+7.2f) = %-+7.2f
'
# 1771: getglobal	0
# 1776: pushint		2
# 1781: call		printf
# 1786: pop
# 1787: pushint		0
# 1792: pushstring	'pi (-+7.3f) = %-+7.3f
'
# 1820: getglobal	0
# 1825: pushint		2
# 1830: call		printf
# 1835: pop
# 1836: pushint		0
# 1841: pushstring	'pi (-+7.4f) = %-+7.4f
'
# 1869: getglobal	0
# 1874: pushint		2
# 1879: call		printf
# 1884: pop
# 1885: pushint		0
# 1890: pushint		0
# 1895: call		println
# 1900: pop
# 1901: halt
