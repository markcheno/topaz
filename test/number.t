


################
# real numbers #
################

pi = 3.1415
println("Default")
printf("pi (f)     = %f\n",pi)
println

println("right justified")
printf("pi (7.0f)  = %7.0f\n",pi)
printf("pi (7.1f)  = %7.1f\n",pi)
printf("pi (7.2f)  = %7.2f\n",pi)
printf("pi (7.3f)  = %7.3f\n",pi)
printf("pi (7.4f)  = %7.4f\n",pi)
println

println("right justified with negative")
printf("-pi (7.0f)  = %7.0f\n",-pi)
printf("-pi (7.1f)  = %7.1f\n",-pi)
printf("-pi (7.2f)  = %7.2f\n",-pi)
printf("-pi (7.3f)  = %7.3f\n",-pi)
printf("-pi (7.4f)  = %7.4f\n",-pi)
println

println("right justified with plus")
printf("pi (+7.0f) = %+7.0f\n",pi)
printf("pi (+7.1f) = %+7.1f\n",pi)
printf("pi (+7.2f) = %+7.2f\n",pi)
printf("pi (+7.3f) = %+7.3f\n",pi)
printf("pi (+7.4f) = %+7.4f\n",pi)
println

println("left justified")
printf("pi (-7.0f)  = %-7.0f\n",pi)
printf("pi (-7.1f)  = %-7.1f\n",pi)
printf("pi (-7.2f)  = %-7.2f\n",pi)
printf("pi (-7.3f)  = %-7.3f\n",pi)
printf("pi (-7.4f)  = %-7.4f\n",pi)
println

println("left justified with negative")
printf("-pi (-7.0f) = %-7.0f\n",-pi)
printf("-pi (-7.1f) = %-7.1f\n",-pi)
printf("-pi (-7.2f) = %-7.2f\n",-pi)
printf("-pi (-7.3f) = %-7.3f\n",-pi)
printf("-pi (-7.4f) = %-7.4f\n",-pi)
println

println("left justified with plus")
printf("pi (-+7.0f) = %-+7.0f\n",pi)
printf("pi (-+7.1f) = %-+7.1f\n",pi)
printf("pi (-+7.2f) = %-+7.2f\n",pi)
printf("pi (-+7.3f) = %-+7.3f\n",pi)
printf("pi (-+7.4f) = %-+7.4f\n",pi)
println
