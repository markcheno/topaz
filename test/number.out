Default
pi (f)     = 3.1415

right justified
pi (7.0f)  =       3
pi (7.1f)  =     3.1
pi (7.2f)  =    3.14
pi (7.3f)  =   3.141
pi (7.4f)  =  3.1415

right justified with negative
-pi (7.0f)  =      -3
-pi (7.1f)  =    -3.1
-pi (7.2f)  =   -3.14
-pi (7.3f)  =  -3.141
-pi (7.4f)  = -3.1415

right justified with plus
pi (+7.0f) =      +3
pi (+7.1f) =    +3.1
pi (+7.2f) =   +3.14
pi (+7.3f) =  +3.141
pi (+7.4f) = +3.1415

left justified
pi (-7.0f)  = 3      
pi (-7.1f)  = 3.1    
pi (-7.2f)  = 3.14   
pi (-7.3f)  = 3.141  
pi (-7.4f)  = 3.1415 

left justified with negative
-pi (-7.0f) = -3     
-pi (-7.1f) = -3.1   
-pi (-7.2f) = -3.14  
-pi (-7.3f) = -3.141 
-pi (-7.4f) = -3.1415

left justified with plus
pi (-+7.0f) = +3     
pi (-+7.1f) = +3.1   
pi (-+7.2f) = +3.14  
pi (-+7.3f) = +3.141 
pi (-+7.4f) = +3.1415

